src/main/kotlin/com/sourcegraph/cody/chat/AgentChatSession.kt

```diff
@@ -140,15 +148,35 @@ private constructor(
             extensionMessage.isMessageInProgress == false) {
           getCancellationToken().dispose()
         } else {
-          if (lastMessage?.text != null && extensionMessage.chatID != null) {
-            addAssistantResponseToChat(lastMessage.text, lastMessage.displayText)
+
+          if (extensionMessage.chatID != null) {
+            if (prevLastMessage != null) {
+              if (lastMessage?.contextFiles != messages.lastOrNull()?.contextFiles) {
+                val index = restoredMessagesCount + extensionMessage.messages.count() - 2
```

Comment:
I was always confused by that `- 2` but I finally get it.
Won't it be the same (but IMHO more easier to understand) if instead you will use `messages.count() - 1`?

Reply:
`extensionMessage.messages` always contains the full transcript of a given session. 

Hence, 
`messages.count() - 1` is always the index of the last element
and 
`messages.count() - 2` is always the index of the element before the last element 

That's all :smiley: